### PR TITLE
Draft: re-do docs landing page

### PR DESCRIPTION
**Pull request by avital**
Try a first attempt at a full re-haul of the RTD landing page with more explicit calls to action. This is a starting point for future exploration.

See a preview at https://flax--1666.org.readthedocs.build/en/1666/

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,18 +3,73 @@
...
+====
 
-Flax is a neural network library and ecosystem for JAX that is
-designed for flexibility. Flax is in use by a growing community of
-researchers and engineers at Google who happily use Flax for their
-daily research.
+Flax delivers an **end-to-end, flexible, user experience for researchers
+who use JAX with neural networks**. Flax exposes the full power of JAX.
+It is made up of loosely coupled libraries,
+which are showcased with end-to-end integrated guides and examples. |emphasized hyperlink|_
```
Comment (8bitmp3) [inline docs/index.rst:12]: @saeta You feedback would be appreciated.

File: docs/index.rst [hunk 2]
```diff
@@ -3,18 +3,73 @@
...
 
-For a quick introduction and short example snippets, see our `README
-<https://github.com/google/flax/blob/main/README.md>`_.
+.. |emphasized hyperlink| replace:: **Get started now ▶**
+.. _emphasized hyperlink: getting_started.html
+
+.. code:: python
```
Comment (andsteing) [inline docs/index.rst:17]: Could you add a link to a working Colab for this?

File: docs/index.rst [hunk 3]
```diff
@@ -3,18 +3,73 @@
...
+
+   class MLP(nn.Module):
```
Comment (andsteing) [inline docs/index.rst:19]: What about using an example that highlights Flax's strengths a bit better?

The challenge will be to keep it short and understandable. I think it's fine if the reader does not understand all the code, but they should be able to understand what is happening, and ideally be impressed by Flax's brevity and great flexibility to tackle difficult problems...

**Review by andsteing — CHANGES_REQUESTED** [review]
It's great to see this ball rolling...

First round of comments - there will probably be lots of iterations on this PR :)

File: docs/index.rst [hunk 4]
```diff
@@ -3,18 +3,73 @@
...
+====
 
-Flax is a neural network library and ecosystem for JAX that is
-designed for flexibility. Flax is in use by a growing community of
-researchers and engineers at Google who happily use Flax for their
-daily research.
+Flax delivers an **end-to-end, flexible, user experience for researchers
+who use JAX with neural networks**. Flax exposes the full power of JAX.
+It is made up of loosely coupled libraries,
+which are showcased with end-to-end integrated guides and examples. |emphasized hyperlink|_
```
Comment (saeta) [inline docs/index.rst:13]: Silly idea; feel free to use it or ignore it:

```suggestion
Flax delivers a **comprehensive platform** for JAX-based researchers, emphasizing **productivity, and flexibility without compromising the full power of JAX's composable function transformations**. Flax's features enable researchers to scale from a single local CPU/GPU up to massive TPU pods [1](http://todo/link-to-t5x-and-related-documentation).
Flax is made up of loosely coupled libraries,
which are showcased with end-to-end integrated guides and examples. |emphasized hyperlink|_
```

**Comment by marcvanzee** [comment]
Closing this since @cgarciae took this over in #2177.

Thanks for starting this @avital!

**Pull request closed without merging**